### PR TITLE
Allow route binding of slug in addition to routeKeyName

### PR DESCRIPTION
**Pull request by belzaaron**
This pull request adds a method to allow the generated slug to be used along side the default (or overridden) `getRouteKeyName()` method. Simply adding another method on the SlugOptions configuration chain can opt-in to this feature. With the addition of the `allowRouteBinding()` in the config you may now instantly use the slug without worrying about converting the whole system to slugs (and it just feels better to me :P)

I'm struggling to find how I can reliably test this so, please if you have any guidance I would love to hear it!

I know that I usually add this feature as another trait that extends the `HasSlug` from this package but, I would like to put this out to maybe anyone who wants to do this with just the package.

Please let me know how you feel about this and if you have any constructive feedback please, I'm all ears!

### CODE REVIEW

File: src/HasSlug.php [hunk 1]
```diff
@@ -52,6 +52,17 @@ public function generateSlug()
...
     }
 
+    public function resolveRouteBinding($value)
+    {
+        if ($this->getSlugOptions()->routeBinding) {
+            return $this->where($this->getRouteKeyName(), $value)
+                        ->orWhere($this->getSlugOptions()->slugField, $value)
+                        ->first();
+        } else {
+            return parent::resolveRouteBinding($value);
```
Comment (freekmurze) [inline src/HasSlug.php]: Could you move this out of the `if` construct, so we can use the `else`branch. Early returns ftw!

**Comment by freekmurze** [comment]
Very neat feature, thanks. Could you also add a test to make sure that it works correctly?

File: src/HasSlug.php (hunk 1 above)
Comment (belzaaron) [reply src/HasSlug.php]: Hey Freek, I was not exactly sure what you meant by

> so we can use the `else` branch

So in my most recent commit (https://github.com/spatie/laravel-sluggable/pull/115/commits/72b512ede88d4252fc9336207a6e8598a98f5cc9) I just removed the else (so if it was what you meant there yah go). Please let me know if this is not what you are referring to.

**Comment by belzaaron** [comment]
> Very neat feature, thanks. Could you also add a test to make sure that it works correctly?

Awesome! I'm not sure exactly how to add a test for this. My best guess would be to define an api route with bindings then test based off of that. But, I have never used Orchestra testbench. Do you have any resources to where I should learn from to set some tests. I'd love to get this up to your standards!

**Comment by freekmurze** [comment]
Thanks!

**Comment by freekmurze** [comment]
Had to revert this because the tests were failing. Could you open up a new PR with your code.

Sorry about this.

**Comment by belzaaron** [comment]
> Had to revert this because the tests were failing. Could you open up a new PR with your code.
> 
> Sorry about this.

@freekmurze, I'm sorry to hear that! I still cannot get the suite to run at all on my local. So I tried to create the tests using wishful-thinking. This long-closed issue (https://github.com/spatie/laravel-sluggable/issues/47) was never resolved from my observation and experience. This is preventing me from effectively writing tests that will pass here. If I open a new pull request with the changes is there a tag / way you could add to advertise that I'm looking for some help. I'd love to see this feature implemented and just need some help getting there :p..... Thanks again and hoping to hear soon.

**Pull request merged**